istioctl/cmd/proxyconfig.go

```diff
@@ -116,13 +116,13 @@ var stringToLevel = map[string]Level{
 }
 
 var (
-	loggerLevelString = ""
-	reset             = false
+	loggerLevelString         = ""
+	reset                     = false
+	isZtunnelPodKubeClient, _ = kubeClient(kubeconfig, configContext)
```

Comment:
I think the error of `kubeClient` is checked across other commands. Could you also check the error instead of ignoring it?

Reply:
Hey @hanxiaop , I can do that if we want. It may be useful to note that the previous code did not check the error, I'm not sure if that was intended or not.

Reply:
Also, yes `kubeClient` is checked within this same file in every other instance except this one. It does seem more consistent to update this logic to also include an error check. I wonder if it still makes sense to bother abstracting it then :thinking:

Reply:
It looks like you authored the initial code with the ignored error check actually @hanxiaop 😅: https://github.com/istio/istio/commit/4721395704e91a388b8f33f296c64a5c77a906f6

Will update shortly!

Reply:
I think an abstraction still makes sense; perhaps just wrap kubeClient initialization in something similar to a sync.Once?

Reply:
@MorrisLaw Thanks for catching the error. I think I made a mistake in the original code. I was considering whether having or not having the initial client error is acceptable, given that `isZtunnelPod` doesn't depend on the client because if not have, it will fallback to the name guess. However, even if it doesn't rely on the client, the subsequent part of extracing configdump, will still use the client.